.github/workflows/ci.yml

```diff
@@ -0,0 +1,32 @@
+name: Lint Check
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  lint:
+    name: Run ESLint and Prettier
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18 # Update to Node.js 18 or higher
+      
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Run ESLint
+        run: npm run lint
+        env:
+          CI: true
+
+      - name: Run Prettier
+        run: npm run format:check
```

Comment:
Maybe we can indicate that we can check only changed files, here is an action that can help to retrieve changed files https://github.com/tj-actions/changed-files